snap/snapcraft.yaml

```diff
@@ -211,11 +211,14 @@ parts:
     - curl
     override-build: |
       snapcraftctl build
-      cp -r ./* $SNAPCRAFT_PART_INSTALL
+
+      mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
+      mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka/plugins
```

Comment:
**todo:** I think we should make a clear distinction between what is a 'plugin' vs what is a 'libs'.

From Apache Kafka, `libs` is whatever is essential to run the packaged Kafka product. In our case, it's what we're shipping with the Snap/Rock by default to be loaded and ran by Kafka every time.

Confluent uses the term `plugins` for any user-created, manually added extras. These could be custom patches, but more generally they're self-managed custom connectors.

We can't write to the `$SNAP` squashfs, so the charm can't add any 'plugins' there.

I recommend we do the following:

```markdown
- $SNAP
    - opt/kafka/libs
        - core kafka libs
        - prom exporter
        - strimzi oauth
        - any future Canonical patches

- $SNAP_DATA
    - etc/kafka/plugins
        - manual connectors
        - sink/source plugins
        - converters
        - etc
```

Reply:
Uhm, if possible I would prefer to keep a bit of separation between the `core kafka libs` and other stuff (`prom exporter`, `strimzi`). I would not put everything in `libs`. `Charmed-kafka` bundles a number of component on top of Kafka, but that does not mean we should put everything in the same folder. 

Actually, I was (and I still am) a bit divided between (a) using `/opt/kafka/plugins` or (b) use `/opt/prometheus/..` and `/opt/strimzi/...`

TBH, the more I think about this the more I lean towards (b)